### PR TITLE
Add some docstrings to the old flax.training.common_utils module.

### PR DESCRIPTION
**Pull request by levskaya**
Add docs to old `training.common_utils` per #729

### CODE REVIEW

File: flax/training/common_utils.py [hunk 1]
```diff
@@ -21,31 +25,78 @@
...
 
 def shard_prng_key(prng_key):
-  # PRNG keys can used at train time to drive stochastic modules
-  # e.g. DropOut. We would like a different PRNG key for each local
-  # device so that we end up with different random numbers on each one,
-  # hence we split our PRNG key and put the resulting keys into the batch
+  """Helper to shard (aka split) a PRNGKey for use with pmap'd functions.
+
+   PRNG keys can used at train time to drive stochastic modules
+   e.g. DropOut. We would like a different PRNG key for each local
```
Comment (cgarciae) [inline flax/training/common_utils.py:45]:
```suggestion
   e.g. Dropout. We would like a different PRNG key for each local
```

Comment (levskaya) [reply flax/training/common_utils.py]: fixed

File: flax/training/common_utils.py [hunk 2]
```diff
@@ -21,33 +25,80 @@
...
 
 def shard_prng_key(prng_key):
-  # PRNG keys can used at train time to drive stochastic modules
-  # e.g. DropOut. We would like a different PRNG key for each local
-  # device so that we end up with different random numbers on each one,
-  # hence we split our PRNG key and put the resulting keys into the batch
-  return jax.random.split(prng_key, num=jax.local_device_count())
+  """Helper to shard (aka split) a PRNGKey for use with pmap'd functions.
 
+   PRNG keys can used at train time to drive stochastic modules
```
Comment (cgarciae) [inline flax/training/common_utils.py:44]: These lines have an additional 1 space of indentation so they are interpreted similar to markdowns `>`:
Screenshot from 2022-08-04 19-28-40

File: flax/training/common_utils.py [hunk 3]
```diff
@@ -21,33 +25,80 @@
...
+  """
+  return jax.random.split(prng_key, num=jax.local_device_count())
 
 
 def stack_forest(forest):
+  """Helper function to stack the leaves of a sequence of pytrees.
+
+  Args:
+    forest: a sequence of pytrees (e.g tuple or list) of matching structure
+    whose leaves are arrays with individually matching shapes.
```
Comment (cgarciae) [inline flax/training/common_utils.py:62]: Missing 2 spaces before "whose" to continue the doc for the current argument (`forest`) in a new line.
Screenshot from 2022-08-04 19-32-22

File: flax/training/common_utils.py [hunk 4]
```diff
@@ -21,33 +25,80 @@
...
   stack_args = lambda *args: np.stack(args)
   return jax.tree_util.tree_map(stack_args, *forest)
 
 
 def get_metrics(device_metrics):
+  """Helper utility for pmap, gathering replicated timeseries metric data.
+
+  Args:
+   device_metrics: replicated, device-resident pytree of metric data,
+   whose leaves are presumed to be a sequence of arrays recorded over time.
```
Comment (cgarciae) [inline flax/training/common_utils.py:76]: Missing 2 spaces, same as previous comment:
Screenshot from 2022-08-04 19-35-10

File: flax/training/common_utils.py [hunk 5]
```diff
@@ -21,33 +25,80 @@
...
+  """
   # We select the first element of x in order to get a single copy of a
   # device-replicated metric.
   device_metrics = jax.tree_util.tree_map(lambda x: x[0], device_metrics)
   metrics_np = jax.device_get(device_metrics)
   return stack_forest(metrics_np)
+
+
+def onehot(labels, num_classes, on_value=1.0, off_value=0.0):
+  """"Create a dense one-hot version of an indexed array.
```
Comment (cgarciae) [inline flax/training/common_utils.py:89]: Remove additional `"` 
```suggestion
  """Create a dense one-hot version of an indexed array.
```

File: flax/training/common_utils.py (hunk 2 above)
Comment (levskaya) [reply flax/training/common_utils.py]: oops, thanks

**Comment by cgarciae** [comment]
Hey! Apart from the previous comments, you should also add an entry with `flax.common_utils` to the `toctree` in `docs/api_reference/index.rst`.

**Comment by levskaya** [comment]
but common_utils.py is just a small subset of training, so probably doesn't belong in toc as its own entry

**Review by cgarciae — APPROVED** [review]
LGTM!

**Pull request merged**